src/Voron/Data/Fixed/FixedSizeIterators.cs

```diff
@@ -237,9 +237,6 @@ public bool MoveNext()
             {
                 AssertNoChanges();
 
-                if (_currentPage == null)
```

Comment:
why remove this?

Reply:
In order to _not_ throw on `MoveNext()` if we already iterated over all values. See the following test:
https://github.com/ravendb/ravendb/pull/8039/files#diff-e373029fe165f4fccd9ea5c3a9ea4b94R134

Now the behavior is consistent with our other iterators and .net framework enumerators.

It could be thrown during compaction, note `MoveNext()` calls inside `while` loops:
https://github.com/ravendb/ravendb/blob/791dec72ff727ca72ce82c6001d1774fe6448664/src/Voron/Impl/Compaction/StorageCompaction.cs#L228-L240

When first `MoveNext` call returned `false` then the second one thew this exception